js/client-battle.js

```diff
@@ -1110,4 +1098,44 @@
 		}
 	});
 
-}).call(this, jQuery);
+	var BattleOptionsPopup = this.BattleOptionsPopup = Popup.extend({
+		initialize: function (data) {
+			this.battle = data.battle;
+			var buf = '<p><strong>In this battle</strong></p>';
+			buf += '<p><label class="optlabel"><input type="checkbox" name="ignorespects" /> Ignore Spectators</label></p>';
+			buf += '<p><label class="optlabel"><input type="checkbox" name="ignoreopp" /> Ignore Opponent</label></p>';
+			buf += '<p><strong>All battles</strong></p>';
+			buf += '<p><label class="optlabel"><input type="checkbox" name="ignorenicks"' + (Tools.prefs('ignorenicks') ? ' checked' : '') + ' /> Ignore nicknames</label></p>';
+			buf += '<p><button name="close">Close</button></p>';
+			this.$el.html(buf);
+		},
+		events: {
+			'change input[name=ignorespects]': 'toggleIgnoreSpects',
+			'change input[name=ignorenicks]': 'toggleIgnoreNicks',
+			'change input[name=ignoreopp]': 'toggleIgnoreOpponent',
+		},
+		toggleIgnoreSpects: function (e) {
+			this.battle.ignoreSpects = !!e.currentTarget.checked;
+			this.battle.add('Spectators ' + (this.battle.ignoreSpects ? '' : 'no longer ') + 'ignored.');
+		},
+		toggleIgnoreNicks: function (e) {
+			this.battle.ignoreNicks = !!e.currentTarget.checked;
+			Tools.prefs('ignorenicks', this.battle.ignoreNicks);
+			this.battle.add('Nicknames ' + (this.battle.ignoreNicks ? '' : 'no longer ') + 'ignored.');
+			this.toggleNicknames(this.battle.mySide);
+			this.toggleNicknames(this.battle.yourSide);
+		},
+		toggleIgnoreOpponent: function (e) {
+			this.battle.ignoreOpponent = !!e.currentTarget.checked;
+			this.battle.add('Opponent ' + (this.battle.ignoreOpponent ? '' : 'no longer ') + 'ignored.');
+		},
+		toggleNicknames: function (side) {
+			for (var i = 0; i < side.active.length; i++) {
+				side.active[i].statbarElem.html(side.getStatbarHTML(side.active[i], true));
+				side.updateStatbar(side.active[i], true, true);
+			}
+			side.updateSidebar();
+		},
+	});
+
+}).call(this, jQuery);
```

Comment:
No, it currently has a trailing linebreak, which you lost.

Trailing linebreaks are bad form; we should probably update eslint to warn about them.